components/generic/CheckBox/CheckBox.tsx

```diff
@@ -0,0 +1,30 @@
+import { ChangeEventHandler } from 'react';
+
+type CheckBoxProps = {
+  name: string;
+  label: string;
+  description?: string;
+  isChecked: boolean;
```

Comment:
```suggestion
  checked: boolean;
```

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/checkbox#attr-checked
"checked" collides with a default attribute. I think "isChecked" is a more fitting name here.

Reply:
It is desired here, since the name is matched and we use it to its original purpose - to indicate whether it's checked or not. It was a different thing with button's `name` I believe.